hazelcast-wm/src/main/java/com/hazelcast/web/SessionCleanUpService.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.web;
+
+import com.hazelcast.instance.OutOfMemoryErrorDispatcher;
+import com.hazelcast.map.EntryProcessor;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An internal service to clean invalid session items
+ * from internal IMap
+ */
+public class SessionCleanUpService {
+
+    private static final long FIXED_DELAY = 60;
+    private static final long FIXED_DELAY1 = 60;
+    private final String name;
+    private final ScheduledExecutorService executor;
+    private final Queue<KeyEntryProcessorPair> failQueue = new ConcurrentLinkedQueue<KeyEntryProcessorPair>();
+
+    public SessionCleanUpService(final String name) {
+        this.name = name;
+        executor = Executors.newSingleThreadScheduledExecutor(new CleanupThreadFactory());
+    }
+
+
+    public void registerHazelcastInstanceDelegate(final HazelcastInstanceDelegate delegate) {
+        executor.scheduleWithFixedDelay(new Runnable() {
+            public void run() {
+                while (failQueue.isEmpty()) {
+                    KeyEntryProcessorPair pair = failQueue.poll();
```

Comment:
rather than `poll`, i prefer `peek`  then `remove`.. because in the case of `poll` then `failure` then `offer`  you just change order of the events actually.. and it can be problematic in some cases.